core/src/main/java/org/apache/iceberg/SnapshotScan.java

```diff
@@ -154,7 +155,17 @@ public CloseableIterable<T> planFiles() {
                   .scanMetrics(ScanMetricsResult.fromScanMetrics(scanMetrics()))
                   .metadata(metadata)
                   .build();
-          context().metricsReporter().report(scanReport);
+          MetricsReporter metricsReporter = context().metricsReporter();
+          if (metricsReporter != null) {
+            try {
+              metricsReporter.report(scanReport);
+            } catch (Exception e) {
+              LOG.warn(
+                  "MetricsReporter threw an exception. metricsReporter={}",
+                  metricsReporter.getClass().getName(),
```

Comment:
I feel like this should all be handled by the metrics reporter implementation, e.g. check out https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/rest/RESTMetricsReporter.java#L62 for example or https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/metrics/MetricsReporters.java#L72

The benefit of that approach is that we don't have to do do this every place metrics reporter attempts to report metrics, it's all internal to the implementation

Reply:
I agree here. The metrics reporter itself has enough context and can log why reporting failed